src/main/scala/io/iohk/ethereum/network/PeerEventBusActor.scala

```diff
@@ -1,20 +1,55 @@
 package io.iohk.ethereum.network
 
+import akka.NotUsed
 import akka.actor.Actor
 import akka.actor.ActorRef
 import akka.actor.Props
+import akka.actor.Terminated
 import akka.event.ActorEventBus
+import akka.stream.OverflowStrategy
+import akka.stream.WatchedActorTerminatedException
+import akka.stream.scaladsl.Source
+import akka.util.Timeout
+
+import scala.concurrent.Future
 
 import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.MessageFromPeer
 import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.PeerDisconnected
 import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.PeerHandshakeSuccessful
 import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier._
 import io.iohk.ethereum.network.handshaker.Handshaker.HandshakeResult
 import io.iohk.ethereum.network.p2p.Message
+import io.iohk.ethereum.network.p2p.messages.Codes
 
 object PeerEventBusActor {
   def props: Props = Props(new PeerEventBusActor)
 
+  /** Handle subscription to the peer event bus via Akka Streams.
+    *
+    * @param peerEventBus ref to PeerEventBusActor
+    * @param messageClassifier specify which messages to subscribe to
+    * @return Source that subscribes to the peer event bus on materialization
+    *   and unsubscribes on cancellation. It will complete when the event bus
+    *   actor terminates.
+    *
+    *   Note:
+    *   - subscription is asynchronous so it may miss messages when starting.
+    *   - it does not complete when a specified peerId disconnects.
+    */
+  def messageSource(peerEventBus: ActorRef, messageClassifier: MessageClassifier): Source[MessageFromPeer, NotUsed] =
+    Source
+      .fromMaterializer { (mat, _) =>
+        import mat.executionContext
+        val (actorRef, src) = Source
+          .actorRef[MessageFromPeer](1, OverflowStrategy.fail)
```

Comment:
this `actorRef` is deprecated. Have you tried to use the other variant they mention in documentation?

Reply:
Missed that, will have a look. 👍